src/Tasks/Microsoft.Common.CrossTargeting.targets

```diff
@@ -46,6 +46,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <!-- indicate to caller that project is RID agnostic so that a global property RuntimeIdentifier value can be removed -->
         <IsRidAgnostic>false</IsRidAgnostic>
         <IsRidAgnostic Condition=" '$(RuntimeIdentifier)' == '' and '$(RuntimeIdentifiers)' == '' ">true</IsRidAgnostic>
+        <!-- Extract necessary information for SetPlatform negotiation -->
+        <IsVcxOrNativeProj Condition="'%(Extension)' == '.vcxproj' or '%(Extension)' == '.nativeproj'">true</IsVcxOrNativeProj>
```

Comment:
Could this be more descriptive, and conditioned on a property or something rather than extension? In the project itself we should be able to know whether it's a C++ project.

Reply:
Good timing, I actually _just_ caught this as a bug. Using metadata here causes issues with batching over each project reference and we end up with extra items being returned.

This is now using $(MSBuildProjectExtension) to check extension. @yuehuang010 does such a "I'm a cpp project" property exist?